### PR TITLE
Plugin welcome

### PR DESCRIPTION
**Pull request by Shaun2D2**

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -36,3 +36,6 @@ JIRA_PASSWORD=
 
 # Optional format specifier for reviewers plugin comment. Can also be configured on a per-repo basis in .pullierc
 REVIEWERS_COMMENT_FORMAT="Hi! I have requested reviews from a few of this repository's maintainers: %s.\n**DO NOT MERGE** this PR without two ✅ reviews (with at least one ✅ from the maintainers)"
+
+# Optional message to greet your new contributors
+WELCOME_MESSAGE="Hey there, thanks for contributing to the project!"
```
Comment (decompil3d) [inline .env.example]:
```suggestion
WELCOME_MESSAGE="Hey there, thanks for contributing to the project!"

```

File: plugins/welcome/index.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+   * Process a PR and check to see if we have a new friend
+   *
+   * @memberof WelcomePlugin
+   * @public
+   * @override
+   * @param {ProbotContext} context webhook context
+   * @param {Commenter} commenter Commenter
+   */
+  async processRequest(context, commenter) {
+    const message = process.env.WELCOME_MESSAGE || 'Thanks for making a contribution to the project!';
```
Comment (decompil3d) [inline plugins/welcome/index.js]: You should follow the pattern used in the reviewers plugin instead:

- Pull in any default message in the plugin constructor (so you don't have to do this on every process request)
- If there is no message specified, perhaps just early return and don't even post a comment?
- Check the `config` parameter for an override message at the org or repo level (see https://github.com/godaddy/pullie/blob/master/plugins/reviewers/index.js#L68)

File: plugins/welcome/index.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  /**
+   * Whether this plugin processes edit actions
+   * @public
+   * @override
+   * @returns {Boolean} Whether this plugin processes edit actions
+   */
+  get processesEdits() {
+    return false;
+  }
+
```
Comment (decompil3d) [inline plugins/welcome/index.js]: This is default, so you can omit it altogether.

File: test/unit/plugins/welcome.test.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    };
+  });
+
+  it('is a constructor', function () {
+    assume(WelcomePlugin).is.a('function');
+    assume(WelcomePlugin).has.length(0);
+    assume(welcomePlugin).is.an('object');
+  });
+
+  it('processes edits', function () {
```
Comment (decompil3d) [inline test/unit/plugins/welcome.test.js]:
```suggestion
  it('does not process edits', function () {
```

File: .env.example (hunk 1 above)
Comment (Shaun2D2) [reply .env.example]: did not see any change here

File: plugins/welcome/index.js (hunk 2 above)
Comment (Shaun2D2) [reply plugins/welcome/index.js]: updated based on feedback, added class property set in constructor from the .env var, am now bailing early if I don't have a message in either .env or in config, added checks for config passed from org or repo.

File: plugins/welcome/index.js (hunk 3 above)
Comment (Shaun2D2) [reply plugins/welcome/index.js]: burned this

File: test/unit/plugins/welcome.test.js (hunk 4 above)
Comment (Shaun2D2) [reply test/unit/plugins/welcome.test.js]: fixed tests

File: .env.example (hunk 1 above)
Comment (decompil3d) [reply .env.example]: I added a line break afterwards

File: plugins/welcome/index.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      state: 'all',
+      creator: context.payload.pull_request.user.login
+    }));
+
+    // get all the PRs by the contributor
+    const pullRequests = response.data.filter(data => data.pull_request);
+
+    // if we only have one, then lets welcome them
+    if (pullRequests.length === 1) {
+      commenter.addComment(message, Commenter.priority.Low);
```
Comment (decompil3d) [inline plugins/welcome/index.js]: I think it'd be nice to make this comment a high priority so it shows up at the top of the comment list -- seems like a good opening to Pullie's comment.

```suggestion
      commenter.addComment(message, Commenter.priority.High);
```

File: plugins/welcome/index.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+   */
+  /**
+   * Process a PR and check to see if we have a new friend
+   *
+   * @memberof WelcomePlugin
+   * @public
+   * @override
+   * @param {ProbotContext} context webhook context
+   * @param {Commenter} commenter Commenter
+   * @param {Object} config Configuration for this plugin
```
Comment (decompil3d) [inline plugins/welcome/index.js]: JSDoc nit

```suggestion
   * @param {Object} [config] Configuration for this plugin
   * @param {string} [config.welcomeMessage] Org or repo-level configured welcome message for new contributors
```

**Pull request merged**